app/Console/Commands/CrawlPets.php

```diff
@@ -0,0 +1,306 @@
+    {
+        $this->removeUnnecessaryNodes($crawler);
+
+        $bodyCrawler = $crawler->filter("body");
+
+        return trim($bodyCrawler->text());
+    }
+
+    protected function getSvgLabelsFromWebpage(Crawler $crawler): array
+    {
+        return $crawler->filter("svg[aria-label]")->each(fn(Crawler $node) => [
+            "aria" => trim((string)$node->attr("aria-label")),
+            "class" => trim($node->attr("class") ?? ""),
+        ]);
+    }
+
+    protected function getImageAltsFromWebpage(Crawler $crawler): array
+    {
+        return $crawler->filter("img[alt]")->each(fn(Crawler $node) => [
+            "alt" => trim((string)$node->attr("alt")),
+            "class" => trim($node->attr("class") ?? ""),
+            "title" => trim($node->attr("title") ?? ""),
+        ]);
+    }
+
+    protected function getIconsFromWebpage(Crawler $crawler): array
+    {
+        return $crawler->filter('i[class*="icon-"], span[class*="icon-"], span.check, i.check')
+            ->each(fn(Crawler $node) => [
+                "class" => trim($node->attr("class") ?? ""),
+                "title" => trim($node->attr("title") ?? ""),
+                "aria" => trim($node->attr("aria-label") ?? ""),
+                "data" => trim($node->attr("data-field") ?? ""),
+            ]);
+    }
+
+    protected function isLikelyAPetPage(string $body, ?int $threshold = null): bool
+    {
+        $body = html_entity_decode($body, ENT_QUOTES | ENT_HTML5);
+        $bodyLower = mb_strtolower($body);
+        $bodyAscii = Str::ascii($bodyLower);
+
+        $petKeywords = array_values(array_filter(array_unique(config("crawl_keywords.pet_keywords", []))));
+        $requiredKeywords = array_values(array_filter(array_unique(config("crawl_keywords.required_keywords", []))));
+
+        $allKeywords = array_merge($petKeywords, $requiredKeywords);
+
+        $allKeywordsLower = array_map(fn($k) => mb_strtolower($k), $allKeywords);
+        $allKeywordsAscii = array_map(fn($k) => Str::ascii(mb_strtolower($k)), $allKeywordsLower);
+
+        if ($threshold === null) {
+            $threshold = min(5, count($allKeywords));
+        }
+
+        $score = 0;
+        $matchedKeywords = [];
+
+        foreach ($allKeywordsLower as $i => $kwLower) {
+            if ($kwLower === "") {
+                continue;
+            }
+
+            if ((stripos($bodyLower, $kwLower) !== false || stripos($bodyAscii, $allKeywordsAscii[$i]) !== false)
+                && !in_array($kwLower, $matchedKeywords, true)
+            ) {
+                $score++;
+                $matchedKeywords[] = $kwLower;
+            }
+        }
+
+        return $score >= $threshold;
+    }
+
+    protected function checkIfUrlContainAntiKeywords(string $url): bool
+    {
+        $antiKeywords = config("crawl.anti_keywords");
+
+        return collect($antiKeywords)->contains(fn($keyword) => stripos($url, $keyword) !== false);
+    }
+}
```

Comment:
This class is too big. I would move it to specialized general service class and think about extracting some methods to separate classes.